extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSTable.scala

```diff
@@ -68,8 +68,13 @@ class TPCDSTable(tbl: String, scale: Int, options: CaseInsensitiveStringMap)
   override def capabilities(): util.Set[TableCapability] =
     Set(TableCapability.BATCH_READ).asJava
 
-  override def newScanBuilder(options: CaseInsensitiveStringMap): ScanBuilder = {
-    new TPCDSBatchScan(tpcdsTable, scale, schema)
+  override def newScanBuilder(scanOpts: CaseInsensitiveStringMap): ScanBuilder = {
+    val map = new util.HashMap[String, String](scanOpts)
+    options.asScala.foreach {
+      case (k, v) => map.putIfAbsent(k, v)
+    }
+    val newOpts = new CaseInsensitiveStringMap(map)
```

Comment:
It's good to consider both SparkConf and ReadOption, can we extract it such as `org.apache.iceberg.spark.SparkConfParser`?

Reply:
Sounds good, I will try it later.